clients/go/gengo.go

```diff
@@ -93,241 +85,63 @@ type WagClient struct {
 
 var _ Client = (*WagClient)(nil)
 
+// New creates a new client. The base path, logger, and http transport are configurable.
+// The logger provided should be specifically created for this wag client. If tracing is required,
+// provide an instrumented transport using the wag clientconfig module. If no tracing is required, pass nil to use
+// the default transport.
+func New(basePath string, logger wcl.WagClientLogger, transport *http.RoundTripper) *WagClient {
```

Comment:
Total nit: I feel a little weird about basePath as a name because we over use it so much in wag talking about dir base paths etc.  serviceName seems more descriptive/relevant to me.

Reply:
I'm a little hesitant to change the param name since its been around for a while and people are pretty familiar with it. Also I dont think that serviceName is correct here - for ex. [here](https://github.com/Clever/app-view-service/blob/master/main.go#L177-L181). basePath is meant to be a path in this [format](https://github.com/Clever/discovery-go/blob/master/discovery.go#L42)